sjasm/io_trd.cpp

```diff
@@ -195,8 +196,32 @@ int TRD_AddFile(char* fname, char* fhobname, int start, int length, int autostar
 			hdr[0x09] = (unsigned char)(start & 0xff);
 			hdr[0x0a] = (unsigned char)(start >> 8);
 		}
+}
+*/
+
+	memset(hdr,' ',9);
+	i = strlen(fhobname);
+	if (i > 1)
+	{
+		char *ext = strrchr(fhobname, '.');
+		if (ext && ext[1])
+		{
+			hdr[8] = ext[1];
+			*ext = '\0';
+			i = strlen(fhobname);
+		}
+	}
+	memcpy(hdr, fhobname, std::min(i,8));
+
+	if (hdr[8] == 'B')	{
+		hdr[0x09] = (unsigned char)(length & 0xff);
+		hdr[0x0a] = (unsigned char)(length >> 8);
+	} else	{
+		hdr[0x09] = (unsigned char)(start & 0xff);
+		hdr[0x0a] = (unsigned char)(start >> 8);
 	}
 
+
```

Comment:
try to avoid pure whitespace changes of old source (or put them into separate commit focused on restyling old code, to make them separate from code changes)